thorlcr/mfilemanager/thmfilemanager.cpp

```diff
@@ -145,7 +145,48 @@ class CFileManager : public CSimpleInterface, implements IThorFileManager
         return max;
     }
 
-    void remove(IDistributedFile &file, IUserDescriptor &userDesc)
+    template<typename T>
+    T blockReportFunc(CJobBase &job, std::function<T(unsigned timeout)> func, unsigned timeout, const char *msg)
+    {
+        // every minute timeout, and set workunit state to blocked for NN minutes
+        WUState wuidState = WUStateUnknown;
+        CTimeMon tm(timeout);
+        if (timeout >= 60000)
+            timeout = 60000;
+        while (true)
+        {
+            try
+            {
+                T ret = func(timeout);
+                if (WUStateUnknown != wuidState)
+                {
+                    Owned<IWorkUnit> workunit = &job.queryWorkUnit().lock();
+                    workunit->setState(wuidState);
```

Comment:
@mckellyln - No, it is resetting to the state it had before (as recorded by line 179)

e.g. it will be in state 'running', 179 is reached because of a timeout and it records that state (running) before setting the state to blocked. Then when/if it succeeds this code (above) is restoring it back to the original state (running).